### PR TITLE
support download with perl file::fetch

### PR DESCRIPTION
**Pull request by h0tw1r3**
minimal OS installs without wget or curl may not have perl LWP::Simple either, but perl File::Fetch is often available.

### CODE REVIEW

**Review by bastelfreak — APPROVED** [review]

File: tasks/install_shell.sh [hunk 1]
```diff
@@ -434,13 +440,33 @@ do_perl() {
...
+
+# do_perl_ff URL FILENAME
+do_perl_ff() {
+  info "Trying perl (File::Fetch)..."
+  run_cmd "perl -e 'use File::Fetch; use File::Copy; my \$ff = File::Fetch->new(uri => \$ARGV[0]); my \$outfile = \$ff->fetch() or die \$ff->server; copy(\$outfile, \$ARGV[1]) or die \"copy failed: \$!\"; unlink(\$outfile) or die \"delete failed: \$!\";' '$1' '$2' 2>>$tmp_stderr"
+  rc=$?
+
+  # check for 404
+  grep "HTTP response: 404" $tmp_stderr 2>&1 >/dev/null
+  if test $? -eq 0 ; then
```
Comment (kenyon) [inline tasks/install_shell.sh:459]:
```suggestion
  if grep -q "HTTP response: 404" $tmp_stderr ; then
```

File: tasks/install_shell.sh [hunk 2]
```diff
@@ -67,13 +67,19 @@ assert_unmodified_apt_config() {
...
 
 # Check whether perl and LWP::Simple module are installed
-exists_perl() {
-  if perl -e 'use LWP::Simple;' >/dev/null 2>&1
-  then
+exists_perl_lwp() {
+  if perl -e 'use LWP::Simple;' >/dev/null 2>&1 ; then
+    return 0
+  fi
+  return 1
```
Comment (kenyon) [inline tasks/install_shell.sh:74]: These can be simplified:

```suggestion
  perl -e 'use LWP::Simple;' >/dev/null 2>&1
```

File: tasks/install_shell.sh (hunk 1 above)
Comment (h0tw1r3) [reply tasks/install_shell.sh:459]: There are so many things that could be improved in this script.
I generally kept the same format as the existing perl function to make it easy to review and approve.

File: tasks/install_shell.sh (hunk 2 above)
Comment (h0tw1r3) [reply tasks/install_shell.sh:74]: I generally kept the same format as the existing perl function to make it easy to review and approve.

Comment (h0tw1r3) [reply tasks/install_shell.sh:74]: The entire script is littered with this type of syntax. Happy to open a separate PR to "write" all the wrongs.

**Review by mhashizume — APPROVED** [review]
Thanks @h0tw1r3 !

**Pull request merged**